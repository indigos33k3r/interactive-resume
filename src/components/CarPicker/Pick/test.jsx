import React from 'react'
import {shallow} from 'enzyme'

describe('CarPickerPick', () => {
  it('renders <CarPickerPick />', () => {
    const component = render()

    expect(component).toMatchSnapshot()
  })

  context('when <button.car-picker-pick__button /> was clicked', () => {
    it('calls #handleButtonClick', () => {
      const pickCarHandler = jest.fn()
      const component = render({pickCarHandler})
      component.find('.car-picker-pick__button').prop('onClick')()

      expect(pickCarHandler).toHaveBeenCalledWith('batmobil')
    })
  })

  function render(props = {}) {
    const defaultProps = {
      pickCarHandler: () => {},
      pickerClass: 'batmobil'
    }
    const CarPickerPick = require('.').default

    return shallow(<CarPickerPick {...defaultProps} {...props} />)
  }
})
