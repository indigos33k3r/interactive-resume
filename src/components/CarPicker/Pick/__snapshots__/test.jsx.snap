// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarPickerPick renders <CarPickerPick /> 1`] = `
<div
  className="car-picker-pick"
>
  <div
    className="car-picker-pick__box"
  >
    <div
      className="car-picker-pick__graphic"
    >
      <div
        className="car-picker-pick__batmobil"
      />
    </div>
    <div
      className="car-picker-pick__text"
    >
      90s Batmobil
    </div>
  </div>
  <button
    className="car-picker-pick__button"
    onClick={[Function]}
  >
    Batman!
  </button>
</div>
`;
